README.md

```diff
@@ -162,6 +162,57 @@ GET /api/resources/get-dependent-workloads?id=x86-ubuntu-18.04-img
 ]
 ```
 
+### 5. List All Resources by gem5 Version
+
+**Endpoint**: `GET /api/resources/list-all-resources`
+
+Retrieve all resources that are compatible with a specific gem5 version. The endpoint performs prefix matching, so a version like `25.0.0.1` will match resources that have `25.0` or `25.0.0` in their `gem5_versions` field.
+
+**Parameters**:
+
+- `gem5-version` (required): The gem5 version to match against (e.g., "23.0", "25.0.0.1")
+
+**Examples**:
+
+```bash
+# Get all resources compatible with gem5 version 23.0
+GET /api/resources/list-all-resources?gem5-version=23.0
+
+# Get all resources compatible with gem5 version 25.0.0.1
+# This will match resources with gem5_versions containing "25", "25.0", "25.0.0", or "25.0.0.1"
+GET /api/resources/list-all-resources?gem5-version=25.0.0.1
+```
+
+**Response Format**:
+
+```json
+[
+  {
+    "id": "riscv-ubuntu-20.04-boot",
+    "resource_version": "3.0.0",
+    "category": "workload",
+    "architecture": "RISCV",
+    "gem5_versions": ["23.0", "22.1", "22.0"],
+    // ... other resource fields
+  },
+  {
+    "id": "arm-hello64-static",
+    "resource_version": "1.0.0",
+    "category": "binary",
+    "architecture": "ARM",
+    "gem5_versions": ["23.0", "22.0"],
+    // ... other resource fields
+  }
+]
+```
+
+**Notes**:
+
+- Uses prefix matching: a parameter like `25.0.0.1` matches resources with `25.0`, `25.0.0`, or `25.0.0.1` in their `gem5_versions` field
```

Comment:
Is it not simply the case that we just match on the Major version compatibility?

If so can we just state that? L.e., that two versions with the same major version will return the same list